### PR TITLE
Added leftEitherT and rightEitherT for EitherT

### PR DESCRIPTION
**Pull request by etorreborre**

### CODE REVIEW

**Comment by nhibberd** [comment]
Now that we have either, should we had a transformers dependency with some version strictness? @markhibberd

File: p.cabal [hunk 1]
```diff
@@ -14,8 +14,9 @@ description:           A set of standard preludes for Ambiata projects. \
 
 library
   build-depends:
-                       base                            >= 4.6        && < 5
-                     , semigroups                      == 0.16.*
+                         base                        >= 4.6        && < 5
+                       , either                      >= 4.4.1      && < 5
```
Comment (markhibberd) [inline p.cabal]: This needs to be a bit wider, we use 4.3 in most project.

Comment (etorreborre) [inline p.cabal]: Right, I made a mistake in thinking that starting with 4.3 would prevent me to use 4.4.1 in my project.

**Comment by markhibberd** [comment]
@nhibberd Can try it. Separate change though.

@etorreborre Looking through the knock on effects I think this should be ok. But if it ends up being too painful because of a dep conflict somewhere we can push up into an `x-`. But :shaved_ice: for now.

**Pull request merged**